_posts/2016-11-18-present-vs-any-vs-exists.markdown

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title:  "Present? vs Any? vs Exists?"
+date: 2016-11-18 13:31:00
+categories: ["benchmark", "performance", "rails"]
+author: "mauro-oto"
+---
+
+When working on a Rails project, you may have seen `present?` calls on
+ActiveRecord relationships. This might feel natural, mostly because `present?`
+exists on all objects [via ActiveSupport](http://guides.rubyonrails.org/active_support_core_extensions.html#blank-questionmark-and-present-questionmark), so you expect the relationship to respond to it,
+but it's actually not a very good thing to do. If all we want to do is check if
+a scope exists in our database, there are better ways than using `present?`.
```

Comment:
This is a little weird because **all scopes exist in your database**. 😏 

You probably mean something like this: "If all we want to do is check if the scope returns any results from the database, ... "